### DISCUSSION TITLE
Multiple lines selection in transaction view of the store commerce app D365

### DISCUSSION THREAD

**Discussion by MansoorAhmedArain** [community/community#138997] [Programming Help]
### Body

Hello everyone, Im trying to implement multiple lines selection on transaction view of the store commerce i have implemented the controller which stacking the lines successfuly but its not highlighting the lines in the grid.
The code which im sharing same was working perfectly fine in MPOS but in store commerce its not highlighting the cart view grid lines.

**Here is my code for the controller**

`import { ProxyEntities } from "PosApi/Entities";
import { IExtensionCartViewControllerContext } from "PosApi/Extend/Views/CartView";
import * as CartView from "PosApi/Extend/Views/CartView";
import { ArrayExtensions, StringExtensions } from "PosApi/TypeExtensions";
export default class CartViewController extends CartView.CartExtensionViewControllerBase {
    public static selectedCartLineId: string = StringExtensions.EMPTY;
    public static selectedCartLineIds: string[] = [];
    public static selectedCartLines: ProxyEntities.CartLine[] = [];
    private _selectedCartLines: ProxyEntities.CartLine[];
    public _isProcessingAddItemOrCustomer: boolean;
   
    constructor(context: IExtensionCartViewControllerContext) {
        super(context);
        this.cartLineSelectedHandler = (data: CartView.CartLineSelectedData): void => {
            if (!this._selectedCartLines) {
                this._selectedCartLines = [];
            }
          
            // Update selected cart lines
            this._selectedCartLines = data.cartLines;
            // Add highlight to newly selected lines
            this._selectedCartLines.forEach(cartLine => {
                const element = document.querySelector(`[data-line-id="${cartLine.LineId}"]`);
                if (element) {
                    element.classList.add('win-selected');
                }
                if (CartViewController.selectedCartLineIds.indexOf(cartLine.LineId) === -1) {
                    CartViewController.selectedCartLineIds.push(cartLine.LineId);
                    CartViewController.selectedCartLines.push(cartLine);
                }
            });
            if (ArrayExtensions.hasElements(this._selectedCartLines)) {
                this.context.logger.logError("Constructor ->Length: " + this._selectedCartLines.length);
            }
        };
        this.cartLineSelectionClearedHandler = (): void => {
            this.context.logger.logError("Cleared ->Length: " + (this._selectedCartLines ? this._selectedCartLines.length : 0));
            // Clear all selected lines
            this._selectedCartLines.forEach(cartLine => {
                const element = document.querySelector(`[data-line-id="${cartLine.LineId}"]`);
                if (element) {
                    element.classList.remove('win-selected');
                }
            });
            this._selectedCartLines = [];
            CartViewController.selectedCartLineIds = [];
            CartViewController.selectedCartLines = [];
        };
    }
}`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by jgschmitz** [comment]
Try making sure that the win-selected class is defined in your styles and properly applied to the cart line rows. You might need to create or modify the class to ensure it works with the grid view in store commerce try something like this 

```
.win-selected {
    background-color: #d3d3d3; /* Highlight color */
    /* Add other styles if needed */
}
```
You can also inspect the cart view in the browser’s developer tools to see if the class is being applied properly. Make sure that the data-line-id attribute exists on the elements you're trying to select. You can add a console log to verify the elements being selected and if the cartLine.LineId values are correct.

```
this._selectedCartLines.forEach(cartLine => {
    const element = document.querySelector(`[data-line-id="${cartLine.LineId}"]`);
    console.log(element); // Add this to check the element being selected
    if (element) {
        element.classList.add('win-selected');
    }
});
```
Let me know if that helps!!
